### ISSUE TITLE
Shared folders don't seem to be shared?

### ISSUE THREAD

**Issue by abirck** [realies/soulseek-docker#25]
I shared my music collection but nobody is downloading from me and I got messages from users telling me they were unable to download from me. Soulseek mentions it needs to restart to pick up config changes, could that be the reason? If so is there some way to restart the service. Restarting the docker seems to clear out the config I've set (it asks to me share a folder again, etc..)

**Comment by realies** [comment]
Please see https://github.com/realies/soulseek-docker/issues/23 and https://github.com/realies/soulseek-docker/issues/1.

**Comment by abirck** [comment]
Okay, did some more investigation. Indeed shortening the time period for saving settings means I don't need to log in again when I restart the container but it hasn't solved the issue that I can't upload to anyone.

To test I signed on with another account on another computed and added my docker user to the soulseek user list. I was able to browse files and queue a download but it downloads never started. The files are reported as being in positions 1...N in the queue but the first file never starts.

Looking at the upload tab in the docker soulseek I have it set to 2 upload slots (but have never uploaded to anyone). Have you seen this before?

**Comment by realies** [comment]
https://www.slsknet.org/news/comment/3874#comment-3874

**Comment by abirck** [comment]
I made sure my port was open. When I use another computer to visit the webpage that checks if your port is open (and fill in the port in the settings) it tells me it's open. I also forwarded the port +1 since you're supposed to do that too,

**Comment by abirck** [comment]
Found out soulseek has a diagnostic output tab. It looks like after queueing the upload it sends back a PM_UPLOAD_DENIED message to the requester and the transfer queue says "Denied, file not shared" (and the files appear under the "Shared File Not Found" diagnostic tab). So just to double check I went back and told it to scan my media library again and it found all the same files, no permission issues there. I opened a console on the docker, went to the music folder and they're all owned by the "souseek" user and the "users" group which is the same identity `/app/SoulseekQT` is running as.

**Comment by abirck** [comment]
No answers of yet (still looking around for things to try) but found another user who seems to be experiencing the exact same behavior: https://www.reddit.com/r/Soulseek/comments/cvliqs/shared_file_not_found_issue_with_soulseek_2018130/

**Comment by realies** [comment]
Does it work fine outside of a container on the same network?

**Comment by abirck** [comment]
Yeah, works okay from my Windows box (haven't tried a non-linux container but could if I really needed to).

To try to verify it's not a file permission thing I attached `strace` to the process and watched what files it opened. When I rescanned my music collection I watched it stat/open all my files but when I requested files from myself (via another user from another computer on the network) it didn't even try opening the files. So I think it's hitting some issue before it even attempts to open/send the file. I guess maybe a next step (unless anyone has other ideas) is blow away my config entirely and start anew to make sure it's not something wrong with the config persisted on disk..... I don't know

**Comment by abirck** [comment]
Success! I got my uploads to work.

Previously I had mounted my music at `/music` in the docker container. Since `/data` is the soulseek user home directory I decided to try moving the music to `/data/music` and now uploads work correctly!

No idea why that was necessary, as I mentioned before it could scan all the files fine from `/music` but happy to have figured out how to make it work. Hopefully anybody else that hits this issue can use this to fix. Thanks!
